File: src/edu/PiJAva/entities/Constat.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package edu.PiJAva.entities;

import edu.PiJAva.services.ServiceUser;
import java.util.Date;

/**
 *
 * @author EXPERT
 */
public class Constat {
    private int id;
    private int user_id;
    private String nom;
    private String prenom;
    private String matricule;
    private String lieu;
    private Date date;
    private String image_degats;

    ServiceUser U =new ServiceUser();
    public int getUser_id() {
        return user_id;
    }

    public void setUser_id(int user_id) {
        this.user_id = user_id;
    }
    
    



    public Constat() {
    }

    public Constat(int id, int user_id, String nom, String prenom, String matricule, String lieu, Date date, String image_degats) {
        this.id = id;
        this.user_id = user_id;
        this.nom = nom;
        this.prenom = prenom;
        this.matricule = matricule;
        this.lieu = lieu;
        this.date = date;
        this.image_degats = image_degats;
    }

    public Constat(int user_id, String nom, String prenom, String matricule, String lieu, Date date, String image_degats) {
        this.user_id = user_id;
        this.nom = nom;
        this.prenom = prenom;
        this.matricule = matricule;
        this.lieu = lieu;
        this.date = date;
        this.image_degats = image_degats;
    }

  

    

   

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return U.getById(user_id).getLastname();
    }

    public void setNom(String nom) {
        this.nom = U.getById(user_id).getLastname();;
    }

    public String getPrenom() {
        return U.getById(user_id).getName();
    }

    public void setPrenom(String prenom) {
        this.prenom = U.getById(user_id).getName();
    }

    public String getMatricule() {
        return matricule;
    }

    public void setMatricule(String matricule) {
        this.matricule = matricule;
    }


    public String getImage_degats() {
        return image_degats;
    }

    public void setImage_degats(String image_degats) {
        this.image_degats = image_degats;
    }

    public String getLieu() {
        return lieu;
    }

    public void setLieu(String lieu) {
        this.lieu = lieu;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }
    
     

    
}

File: dev-clan-emna - Copie/src/Entity/Constat.php
<?php

namespace App\Entity;

use App\Repository\ConstatRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;

#[ORM\Entity(repositoryClass: ConstatRepository::class)]
class Constat
{
 #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    #[Groups("constats")]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le champ 'nom' est obligatoire")]
    #[Groups("constats")]
    private ?string $nom = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le champ 'prenom' est obligatoire")]
    #[Groups("constats")]
    private ?string $prenom = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le champ 'matricule' est obligatoire")]
    #[Groups("constats")]
    private ?string $matricule = null;

    #[ORM\Column(type: Types::DATE_MUTABLE)]
    #[Groups("constats")]
    private ?\DateTimeInterface $date = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: "Le champ 'lieu' est obligatoire")]
    #[Groups("constats")]
    private ?string $lieu = null;

    #[ORM\Column(type:"string", length:255, nullable:true)]
    #[Groups("constats")]
    private $image_Degats;

    // #[ORM\ManyToOne(inversedBy: 'constats')]
    // private ?Expert $expert = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(?string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getPrenom(): ?string
    {
        return $this->prenom;
    }

    public function setPrenom(?string $prenom): self
    {
        $this->prenom = $prenom;

        return $this;
    }

    public function getMatricule(): ?string
    {
        return $this->matricule;
    }

    public function setMatricule(?string $matricule): self
    {
        $this->matricule = $matricule;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(?\DateTimeInterface $date): self
    {
        $this->date = $date;

        return $this;
    }

    public function getLieu(): ?string
    {
        return $this->lieu;
    }

    public function setLieu(?string $lieu): self
    {
        $this->lieu = $lieu;

        return $this;
    }

    public function getImageDegats()
    {
        return $this->image_Degats;
    }

    public function setImageDegats($image_Degats)
    {
        $this->image_Degats = $image_Degats;

        return $this;
    }

    // public function getExpert(): ?Expert
    // {
    //     return $this->expert;
    // }

    // public function setExpert(?Expert $expert): self
    // {
    //     $this->expert = $expert;

    //     return $this;
    // }


   


}
